core/chaincode/lifecycle/lifecycle.go

```diff
@@ -697,6 +713,16 @@ func (ef *ExternalFunctions) InstallChaincode(chaincodeInstallPackage []byte) (*
 	}, nil
 }
 
+func (ef *ExternalFunctions) getBuildLock(packageID string) *sync.Mutex {
+	ef.mutex.Lock()
+	defer ef.mutex.Unlock()
+	buildLock, ok := ef.BuildLocks[packageID]
```

Comment:
Up to you, but you could do lazy map initialization here:

```
if ef.BuildLocks == nil {
    ef.BuildLocks = map[string]sync.Mutex{}
}
```

then you could skip the initialization elsewhere, but, not a requirement.

Reply:
I considered this but I wasn't entirely sure if it was acceptable or not. I definitely prefer this over the random empty map initialization all the way out in peer/node/start. Done.